File: servico-musica-java/java.soap/src/main/java/servico/java/soap/controller/MusicaController.java
package servico.java.soap.controller;

import org.springframework.ws.server.endpoint.annotation.Endpoint;
import org.springframework.ws.server.endpoint.annotation.PayloadRoot;
import org.springframework.ws.server.endpoint.annotation.RequestPayload;
import org.springframework.ws.server.endpoint.annotation.ResponsePayload;

import servico.java.soap.generated.GetTodasMusicasRequest;
import servico.java.soap.generated.GetTodasMusicasResponse;
import servico.java.soap.repository.MusicaRepository;
import servico.java.soap.generated.Musica;

@Endpoint
public class MusicaController {

    private static final String NAMESPACE_URI = "http://soap.java.servico/musica";

    private final MusicaRepository musicaRepository;

    public MusicaController(MusicaRepository musicaRepository) {
        this.musicaRepository = musicaRepository;
    }

    @PayloadRoot(
            namespace = NAMESPACE_URI,
            localPart = "getTodasMusicasRequest"
    )
    @ResponsePayload
    public GetTodasMusicasResponse getTodasMusicas(
            @RequestPayload GetTodasMusicasRequest request) {

        GetTodasMusicasResponse response =
                new GetTodasMusicasResponse();

        musicaRepository.findAll().forEach(m -> {

            Musica soapMusica = new servico.java.soap.generated.Musica();

            soapMusica.setId(m.getId());
            soapMusica.setNome(m.getNome());
            soapMusica.setArtista(m.getArtista());

            response.getMusica().add(soapMusica);
        });

        return response;
    }
}

File: servico-musica-kotlin/kotlin.soap/src/main/kotlin/servico/kotlin/soap/controller/MusicaController.kt
package servico.kotlin.soap.controller

import org.springframework.ws.server.endpoint.annotation.Endpoint
import org.springframework.ws.server.endpoint.annotation.PayloadRoot
import org.springframework.ws.server.endpoint.annotation.RequestPayload
import org.springframework.ws.server.endpoint.annotation.ResponsePayload
import servico.kotlin.soap.generated.GetTodasMusicasRequest
import servico.kotlin.soap.generated.GetTodasMusicasResponse
import servico.kotlin.soap.generated.Musica
import servico.kotlin.soap.repository.MusicaRepository

@Endpoint
class MusicaController(private val musicaRepository: MusicaRepository) {

    @PayloadRoot(namespace = NAMESPACE_URI, localPart = "getTodasMusicasRequest")
    @ResponsePayload
    fun getTodasMusicas(
        @RequestPayload request: GetTodasMusicasRequest?
    ): GetTodasMusicasResponse {
        val response: GetTodasMusicasResponse =
            GetTodasMusicasResponse()

        musicaRepository.findAll().forEach { m ->
            val soapMusica = Musica()
            soapMusica.id = m.id!!
            soapMusica.nome = m.nome
            soapMusica.artista = m.artista

            response.musica.add(soapMusica)
        }

        return response
    }

    companion object {
        private const val NAMESPACE_URI = "http://soap.kotlin.servico/musica"
    }
}